eva/readers/decord_reader.py

```diff
@@ -121,6 +132,45 @@ def _read(self) -> Iterator[Dict]:
                             "seconds": math.floor(
                                 video.get_frame_timestamp(frame_id)[0]
                             ),
+                            "audio": np.empty(0),
                         }
                     else:
                         break
+
+    def get_av_frames(self) -> Iterator[Dict]:
+        decord = _lazy_import_decord()
+        if self._sampling_type is not None:
+            err_msg = "Sampling type not supported with audio based queries"
+            logger.exception(err_msg)
+            raise Exception(err_msg)
+        if self._sampling_rate != 1:
+            err_msg = "Sampling rate not supported with audio based queries"
+            logger.exception(err_msg)
```

Comment:
Heads up: We have moved to a newer design for error handling, where we don't raise exceptions but rather use asserts. Have a look at the master codebase for references. Thanks!